### PR TITLE
Feat/router setup

### PR DESCRIPTION
**Pull request by Devynnem**
____ Wrote Tests ____ Implemented ____ Reviewed

## Neccesary checkmarks
- [ ] All Tests are Passing in all environments
- [x] The code will run locally
## Type of change
- [x] New feature
- [ ] Bug Fix
- [ ] Refactor
## Description of Change:

  description:  This PR implements Router and adds navLinks to various now clickable items on the page. We are able to switch between vendor and customer view, as well as see the specific vendor details rendered.
    
## Requested feedback: 
    
  I'd like feedback on... 

## Check the correct boxes

- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes

- [ ] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

- [ ] My code has no unused/commented out code
- [ ] My code has no binding.pry's
- [ ] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "@testing-library/user-event": "^13.5.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
+    "react-router-dom": "^6.14.1",
```
Comment (Pma913) [inline package.json:11]: Nice job implementing the new version of router

File: src/Components/Results/Results.js [hunk 2]
```diff
@@ -2,13 +2,14 @@ import React from 'react'
...
 const Results = ({ vendors, addFav, remFav, favorites }) => {
-  const allVendors = vendors.map((truck, element) =>  {
+  const allVendors = vendors.map((truck) =>  {
     return(
       <TruckCard 
         truck={truck}
         addFav={addFav}
         remFav={remFav}
-        key={element} />
+        key={truck.id}
```
Comment (Pma913) [inline src/Components/Results/Results.js:11]: Thank you for fixing the keys for the vendors

**Review by Pma913 — APPROVED** [review]
Really nice job implementing React Router, clean and easy to understand!

File: src/Components/App/App.js [hunk 3]
```diff
@@ -64,16 +66,32 @@ const fetchData = () => {
...
+            {/* //params.match, need to pass down the id that is clicked as a prop to truck details to element specfic truck info */}
+          </div>
+       } />
+      <Route path="/vendor-view" element={
+          <div>
+            <Header togView={toggleView} />
+            <VendorView vendors={vendors} />
+          </div>
+      } />
+      <Route path="*" element={
```
Comment (emlindvall) [inline src/Components/App/App.js:90]: This reminds me that we need to set up an Error component!

File: src/Components/TruckCard/TruckCard.js [hunk 4]
```diff
@@ -5,6 +5,7 @@ import notDownvoted from '../../assets/notDownvoted.png'
 import notUpvoted from '../../assets/notUpvoted.png'
 import downvoted from '../../assets/downvoted.png'
 import upvoted from '../../assets/upvoted.png'
+import { NavLink } from 'react-router-dom';
```
Comment (emlindvall) [inline src/Components/TruckCard/TruckCard.js:8]: I'm working in this Component at the moment so there's going to be some merge conflicts here, but I'll get it worked out once my CSS PR is up

**Review by emlindvall — APPROVED** [review]

**Pull request merged**